java/server/src/org/openqa/selenium/grid/distributor/local/LocalDistributor.java

```diff
@@ -88,33 +108,125 @@
   private final GridModel model;
   private final Map<NodeId, Node> nodes;
 
+  private final NewSessionQueuer sessionRequests;
+  private final Duration sessionRequestTimeout;
+
   public LocalDistributor(
       Tracer tracer,
       EventBus bus,
       HttpClient.Factory clientFactory,
       SessionMap sessions,
-      Secret registrationSecret) {
+      NewSessionQueuer sessionRequests,
+      Secret registrationSecret,
+      Duration sessionRequestTimeout) {
```

Comment:
Why do we need the `sessionRequestTimeout` as in the constructor? I thought this value was needed when creating the `NewSessionQueue`. Wouldn't this duplicate the value?

I think it would be better to have the configuration of the `sessionRequestTimeout` only in the queue bits of the code.

Reply:
The LocalDistributor needs to know the timeout, so when it receives the request it will be able to reject immediately if it has timed out. The queue simply just stores the session requests.

Reply:
Maybe the queue can see that the request has timed out even before sending the request to the distributor? This would reduce messages and calls, I think.

Reply:
Yes, it can do that actually. Before the event is fired to indicate to the Distributor to remove from the queue, the queue can check if it has timed out. Do you suggest this change happens as part of this PR or a new one? Since a lot of session queue files will get updated.

Reply:
Good point, having it in a different PR is a good idea.

Reply:
Addressed this one https://github.com/SeleniumHQ/selenium/pull/8769